src/v/serde/parquet/encoding.cc

```diff
@@ -156,11 +81,31 @@ iobuf encode_levels(
     return encode_levels_impl(max_value, levels);
 }
 
-iobuf encode_for_stats(boolean_value v) { return encode_plain({v}); }
-iobuf encode_for_stats(int32_value v) { return encode_plain({v}); }
-iobuf encode_for_stats(int64_value v) { return encode_plain({v}); }
-iobuf encode_for_stats(float32_value v) { return encode_plain({v}); }
-iobuf encode_for_stats(float64_value v) { return encode_plain({v}); }
+iobuf encode_for_stats(boolean_value v) {
```

Comment:
How are these used? Are they called once per value? Are they included in the new benchmarks?

Seems like these are going to be more expensive than the encoding itself since they use a one-shot encoder object, which means a fresh iobuf, iobuf moves, etc.

Reply:
AFAIK, these are only used once per page to encode statistics summarizing the values in the page(i.e, min value, max value..). They are included in the microbenchmark's measured region as they are called each time a page is flushed. I didn't bother trying to optimize them as the new version should be pretty similar to the old and it's only used a small number of times compared to the actual value encoding.

Reply:
yeah it's not on the hot path, it's called twice every 512k worth of data then twice again at the end of a row group for each column

Reply:
While values are being added to a page we collect the min and max value. Then once per page when it is flushed the min & max values are serialized with `encode_for_stats`.

Reply:
Thanks, it's clear now!